public/app/features/alerting/unified/rule-list/filter/useSavedSearches.ts

```
@@ -0,0 +1,424 @@
+    return () => {
+      sessionStorage.removeItem(sessionKey);
+    };
+  }, [notifyApp]);
+
+  /**
+   * Persist saved searches to UserStorage.
+   */
+  const persistSearches = useCallback(
+    async (searches: SavedSearch[]): Promise<void> => {
+      try {
+        await userStorage.setItem(SAVED_SEARCHES_STORAGE_KEY, JSON.stringify(searches));
+      } catch (error) {
+        console.error('Failed to save searches:', error);
+        notifyApp.error(
+          t('alerting.saved-searches.error-save-title', 'Failed to save'),
+          t('alerting.saved-searches.error-save-description', 'Your changes could not be saved. Please try again.')
+        );
+        throw error;
+      }
+    },
+    [notifyApp]
+  );
+
+  /**
+   * Save a new search with the given name and query.
+   * Returns a ValidationError if the name is not unique.
+   */
+  const saveSearch = useCallback(
+    async (name: string, query: string): Promise<void | ValidationError> => {
+      // Validate name uniqueness (case-insensitive)
+      const isDuplicate = savedSearches.some((s) => s.name.toLowerCase() === name.toLowerCase());
+      if (isDuplicate) {
+        return {
+          field: 'name',
+          message: t('alerting.saved-searches.error-name-duplicate', 'A saved search with this name already exists'),
+        };
+      }
+
+      const newSearch: SavedSearch = {
+        id: crypto.randomUUID(),
+        name,
+        query,
+        isDefault: false,
+        createdAt: Date.now(),
+      };
+
+      const newSearches = [...savedSearches, newSearch];
+
+      try {
+        await persistSearches(newSearches);
+        setSavedSearches(newSearches);
+
+        // Track analytics
+        trackSavedSearchSave({
+          hasDefault: newSearches.some((s) => s.isDefault),
+          totalCount: newSearches.length,
+        });
+      } catch {
+        // Error already handled in persistSearches
+      }
```

We should avoid swallowing errors